plugin/pkg/auth/authorizer/webhook/webhook.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package webhook implements the authorizer.Authorizer interface using HTTP webhooks.
+package webhook
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/golang/glog"
+
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/authorization/v1beta1"
+	"k8s.io/kubernetes/pkg/auth/authorizer"
+	"k8s.io/kubernetes/pkg/util"
+)
+
+// Ensure Webhook implements the authorizer.Authorizer interface.
+var _ authorizer.Authorizer = (*WebhookAuthorizer)(nil)
+
+type WebhookAuthorizer struct {
+	callback string
+	cli      *http.Client
+}
+
+func New(callback, clientCertFile, clientKeyFile, caFile string) (*WebhookAuthorizer, error) {
```

Comment:
ah, sorry, forgot to add a comment line there. Should be `callbackURL` and look something like this

```
// New creates a new authorization webhook which drives authorization by querying the remote service at callbackURL.
//
// clientCertFile, clientKeyFile and caFile are used to specify client TLS configurations for the returned webhook.
func New(callbackURL, clientCertFile, clientKeyFile, caFile string) (*WebhookAuthorizer, error) {
```